app/src/main/java/com/rkm/tasky/ui/activity/MainActivity.kt

```diff
@@ -1,34 +1,65 @@
-package com.rkm.tasky
+package com.rkm.tasky.ui.activity
 
+import android.annotation.SuppressLint
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
+import androidx.activity.viewModels
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.padding
 import androidx.compose.material3.Scaffold
 import androidx.compose.material3.SnackbarDuration
 import androidx.compose.material3.SnackbarHost
 import androidx.compose.material3.SnackbarHostState
 import androidx.compose.material3.SnackbarResult
-import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
-import com.rkm.tasky.feature.loginscreen.screen.LoginScreenRoot
+import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
 import com.rkm.tasky.feature.snackbar.SnackBarController
 import com.rkm.tasky.navigation.AppNavigation
 import com.rkm.tasky.ui.event.ObserveAsEvents
 import com.rkm.tasky.ui.theme.TaskyTheme
 import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.launch
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
+
+    private val viewModel by viewModels<MainViewModel>()
+
+    @SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        var authState: AuthState by mutableStateOf(AuthState.Loading)
+
+        lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.authState
+                    .onEach { authState = it }
+                    .collect()
+            }
+        }
+
+        installSplashScreen().apply {
+            setKeepOnScreenCondition {
+                when(authState) {
+                    AuthState.Loading -> false
+                    AuthState.NoSessionInfo -> true
+                    AuthState.Error -> true
+                    AuthState.AuthenticatedUser -> true
+                }
+            }
```

Comment:
Observing the auth state above is not necessary. The `setKeepOnScreenCondition` function is called with every frame, so it will always refer to the latest authState from the ViewModel when just calling `viewModel.authState` in that function block